src/etl/impl/LedgerLoader.hpp

```diff
@@ -136,19 +144,25 @@ class LedgerLoader {
             );
         }
 
-        // Remove all but the last NFTsData for each id. unique removes all but the first of a group, so we want to
-        // reverse sort by transaction index
-        std::sort(result.nfTokensData.begin(), result.nfTokensData.end(), [](NFTsData const& a, NFTsData const& b) {
-            return a.tokenID > b.tokenID && a.transactionIndex > b.transactionIndex;
-        });
+        auto const uniqueNftData = [](std::vector<NFTsData>& nfTokensData) {
+            // Remove all but the last NFTsData for each id. unique removes all but the first of a group, so we want
+            // to reverse sort by transaction index
+            std::sort(nfTokensData.begin(), nfTokensData.end(), [](NFTsData const& a, NFTsData const& b) {
+                return a.tokenID > b.tokenID && a.transactionIndex > b.transactionIndex;
+            });
+
+            // Now we can unique the NFTs by tokenID.
+            auto last = std::unique(nfTokensData.begin(), nfTokensData.end(), [](NFTsData const& a, NFTsData const& b) {
+                return a.tokenID == b.tokenID;
+            });
+            nfTokensData.erase(last, nfTokensData.end());
+        };
+
+        uniqueNftData(result.nfTokensData);
+        uniqueNftData(nfTokenURIChanges);
 
-        // Now we can unique the NFTs by tokenID.
-        auto last = std::unique(
-            result.nfTokensData.begin(),
-            result.nfTokensData.end(),
-            [](NFTsData const& a, NFTsData const& b) { return a.tokenID == b.tokenID; }
-        );
-        result.nfTokensData.erase(last, result.nfTokensData.end());
+        // Put uri change at the end to ensure the uri not overwritten
+        result.nfTokensData.insert(result.nfTokensData.end(), nfTokenURIChanges.begin(), nfTokenURIChanges.end());
```

Comment:
what would happen if NFTokenMint and NFTokenModify occur in the same ledger? in that case, would the  uri of NFTokenModify overwrite the NFTokenMint?

Reply:
Yes perm will fail if Modify occurs before Mint, but it's possible that Modify occur after a Mint in the same ledger, although incredibly rare

Reply:
yes the modified URI should be the final state. this would mean that the uri written by the Mint would be overwritten by the Modify, since both would have the same `token_id` and `sequence`. i just want to confirm the behavior